transformer_engine/common/util/cast_kernels.cuh

```diff
@@ -1142,6 +1142,19 @@ void fp8_quantize_arch_ge_100(const Tensor &input, const Tensor *act_input, cons
                                                              workspace, stream);
       break;
     }
+    case NVTE_CURRENT_TENSOR_SCALING: {
+      if (IS_DBIAS) {
+        // zhongboz: should we just ignore IS_ACT here?
+        NVTE_ERROR("Not implemented scaling mode with DBIAS fusion: " +
+                   to_string(output->scaling_mode) + " on GPU with compute capability >= 10.0.");
+      }
+      if (!IS_DACT) {
+        CastVectorizedUnaryKernelLauncher<ParamOP, OP>(input, noop, output, stream);
+      } else {
+        CastVectorizedUnaryGradKernelLauncher<ParamOP, OP>(input, act_input, output, stream);
```

Comment:
This is not correct - DACT (just like ACT) will modify the original input before casting to FP8 so your scale is no longer correct. In those examples the current scaling does not actually work and we need to do higher precision kernel followed by cast instead.

Reply:
As a general point, I don't think we should support any fused CS kernels at this point. That requires computing intermediate outputs in BF16, reducing amax between CTAs (and maybe between GPUs), and casting to FP8. Doing this in a single kernel would be quite challenging, although if we can figure out how then that's an argument for incorporating more CS logic into the API.